lib/galaxy/webapps/galaxy/api/cloud.py

```diff
@@ -3,255 +3,104 @@
-
-        :rtype:     dictionary
-        :return:    Information about the (un)successfully submitted dataset send jobs,
-                    containing the following keys:
-
-                        *   `bucket_name`:                  The name of bucket to which the listed datasets are queued
-                                                            to be sent.
-                        *   `sent_dataset_labels`:          A list of JSON objects with the following key-value pair:
-                            **  `object`:                   The name of object is queued to be created.
-                            **  `job_id`:                   The id of the queued send job.
-
-                        *   `failed_dataset_labels`:        A list of JSON objects with the following key-value pair
-                                                            representing the datasets Galaxy failed to create
-                                                            (and queue) send job for:
-
-                            **  `object`:                   The name of object is queued to be created.
-                            **  `error`:                    A descriptive error message.
-
-        """
-        missing_arguments = []
-        encoded_history_id = payload.get("history_id", None)
-        if encoded_history_id is None:
-            missing_arguments.append("history_id")
-
-        bucket = payload.get("bucket", None)
-        if bucket is None:
-            missing_arguments.append("bucket")
-
-        encoded_authz_id = payload.get("authz_id", None)
-        if encoded_authz_id is None:
-            missing_arguments.append("authz_id")
-
-        if len(missing_arguments) > 0:
-            raise ActionInputError(f"The following required arguments are missing in the payload: {missing_arguments}")
-
-        try:
-            history_id = self.decode_id(encoded_history_id)
-        except exceptions.MalformedId as e:
-            raise ActionInputError(f"Invalid history ID. {e}")
-
-        try:
-            authz_id = self.decode_id(encoded_authz_id)
-        except exceptions.MalformedId as e:
-            raise ActionInputError(f"Invalid authz ID. {e}")
-
-        encoded_dataset_ids = payload.get("dataset_ids", None)
-        if encoded_dataset_ids is None:
-            dataset_ids = None
-        else:
-            dataset_ids = set()
-            invalid_dataset_ids = []
-            for encoded_id in encoded_dataset_ids:
-                try:
-                    dataset_ids.add(self.decode_id(encoded_id))
-                except exceptions.MalformedId:
-                    invalid_dataset_ids.append(encoded_id)
-            if len(invalid_dataset_ids) > 0:
-                raise ActionInputError(
-                    "The following provided dataset IDs are invalid, please correct them and retry. "
-                    "{}".format(invalid_dataset_ids)
-                )
-
+        return SummaryGetObjects(datasets=rtv)
```

Comment:
Please note that this will not directly return a list as the old deprecated endpoint does.
This will rather return something like:
```json
{
    "datasets": [<The list>] 
}
```

Reply:
I am not really sure what the serialize_to_view returns. Because of this, I am not sure how to construct a corresponding pydantic model and would do the same as the old endpoint:

```suggestion
        return rtv
```

Reply:
Sorry, I wasn´t very clear in my comment.

I mean that the old enpoint seems to return a JSON list of (whatever the `serialize_to_view` is returning for each dataset). Something like:

```json
[
    {
        // datasets[0] values returned by serialize_to_view
    },
    ...
    {
        // datasets[n] values returned by serialize_to_view
    }
]
```

While returning `SummaryGetObjects(datasets=rtv)` will result in a JSON response similar to this:

```json
{
    "datasets": [
        {
            // datasets[0] values returned by serialize_to_view
        },
        ...
        {
            // datasets[n] values returned by serialize_to_view
        }
    ]
}
```

You can have a look at what `self.datasets_serializer.serialize_to_view(dataset, view="summary")` is returning in the [DatasetSerializer class](https://github.com/galaxyproject/galaxy/blob/77265db51ed04a2f28ef11e4578a80ebe4ac9e34/lib/galaxy/managers/datasets.py#L234).

I'm not sure if there is already a model in the schema.py with those values, worth checking. Then you also need to change your return type hint in line 64.

Reply:
I think I understood what you meant earlier, but I have badly worded my problem. 

My problem was that `self.datasets_serializer.serialize_to_view(dataset, view="summary")` returns a dictionary and I did not know what this dictionary contained. So it was hard to construct a pydantic model for the items that are appended to rval.

But I think I found what it returns now.